### ISSUE TITLE
Pen Test actions for ADTS

### ISSUE THREAD

**Issue by Andybacon76** [Andybacon76/Unito_Test#21]

**Comment by Andybacon76** [comment]
➤ Jason Bramwell commented:
L:\MSD\ITU\Software\master\srvbuild\Server Configs\Victor\PenTest\20161013_APHA_WebApp.pdf

**Comment by Andybacon76** [comment]
➤ Jason Bramwell commented:
I have completed all of my bits on test and they seem ok. The CSP raised further issues for David so that it not as tight as it ought to be. Will push these changes out to dev and qa next week and then write up before pushing to live

**Comment by Andybacon76** [comment]
➤ Jason Bramwell commented:
Scheduled for Armadillo-Bertie release next week

**Comment by Andybacon76** [comment]
➤ Jason Bramwell commented:
app 'fixes' put in and updated install script to reflect this. Also captured config files to enable easy redeployment.

**Comment by Andybacon76** [comment]
➤ jane archer commented:
List of Actions  and what we have done/ going to do

**Comment by Andybacon76** [comment]
➤ jane archer commented:
done - but Jason run own test and more actions to look at

**Comment by Andybacon76** [comment]
➤ Jason Bramwell commented:
Strip Apache header
yum install mod_security

add to httpd.conf file
<IfModule security2_module>
    SecRuleEngine on
    ServerTokens Full
    SecServerSignature " "
</IfModule>

restart apache



Strip php header
X-Powered-By header can be removed by setting expose_php to “Off” in
the ‘php.ini’ file.



Weak RC4 Cipher - /etc/httpd/conf/httpd.conf
SSLProtocol All -SSLv2 -SSLv3
SSLHonorCipherOrder On
SSLCompression Off
SSLCipherSuite EECDH+ECDSA+AESGCM:EECDH+aRSA+AESGCM:EECDH+ECDSA+SHA384:EECDH+ECDSA+SHA256:EECDH+aRSA+SHA384:EECDH+aRSA+SHA256:EECDH+aRSA+RC4:EECDH:EDH+aRSA:DHE-RSA-CAMELLIA256-SHA:DHE-RSA-CAMELLIA128-SHA:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-SEED-SHA:AES256-SHA256:AES128-SHA256:AES128-SHA:DHE-RSA-DES-CBC3-SHA:DES-CBC3-SHA:!aNULL:!eNULL:!ADH:!EXP:!LOW:!DES:!MD5:!PSK:!SRP:!DSS:!RC4



CSP - https://content-security-policy.com/
Add to virtual host (/etc/httpd/conf.d/xyz.conf
Header set Content-Security-Policy "default-src 'self'; font-src 'self' data:; img-src 'self' data: https://www.google-nalytics.com; script-src 'self' 'unsafe-eval' 'unsafe-inline' https://code.jquery.com www.google-analytics.com ajax.googleapis.com; style-src 'self' 'unsafe-inline';" 



Disable directory listing - https://www.animal-disease-testing.service.gov.uk/icons/
I think as easy as remove /etc/httpd/conf.d/autoindex.conf (I actually renamed this to .old)



Obsolete webpages - https://www.animal-disease-testing.service.gov.uk/icons/README
Might be fixed by above

**Comment by Andybacon76** [comment]
➤ Jason Bramwell commented:
All added to install script

**Comment by Andybacon76** [comment]
➤ Jason Bramwell commented:
All sites score A+ for SSL security now

https://www.ssllabs.com/ssltest/analyze.html?d=dev.animal-disease-testing.service.gov.uk
https://www.ssllabs.com/ssltest/analyze.html?d=test.animal-disease-testing.service.gov.uk
https://www.ssllabs.com/ssltest/analyze.html?d=qa.animal-disease-testing.service.gov.uk
https://www.ssllabs.com/ssltest/analyze.html?d=www.animal-disease-testing.service.gov.uk
